security/tutorials/calico-policy.md

```diff
@@ -0,0 +1,211 @@
+spec:
+  selector: all()
+  types:
+  - Ingress
+  - Egress
+EOF
+```
+
+#### Verify access - denied all ingress and egress
+
+Because all pods in the namespace are now selected, any ingress traffic which is not explicitly allowed by a policy will be denied.
+
+We can see that this is the case by switching over to our "access" pod in the namespace and attempting to access the nginx service.
+
+```bash
+wget -q --timeout=5 nginx -O -
+```
+
+It will return:
+
+```
+wget: bad address 'google.com'
+```
+
+{: .no-select-button}
+
+Next, try to access google.com.
+
+```bash
+wget -q --timeout=5 google.com -O -
+```
+
+It will return:
+
+```
+wget: bad address 'google.com'
+```
+
+{: .no-select-button}
+
+Now that we have the defalt deny Global Calico Network Policy, all ingress / egress traffic is denied _everywhere_.
+
+### 3. Allow kube-system pods to communicate
+
+Kubernetes uses the kube-system namespace to create pods for the kubernetes cluster to function properly. Let's give this namespace wide permissions to allow these specific pods to function normally. It's important to note here that specific Calico Network Policies are namespaced resources that applies to workloads in that namespace whereas the Calico Global Network Policy is a non-namespaced resource and can be applied to any kind of endpoint (pods, VMs, host interfaces) independent of namespace.
+
+```bash
+calicoctl create -f - <<EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: allow-kube-system
+  namespace: kube-system
+spec:
+  selector: all()
+  types:
+  - Ingress
+  - Egress
+  ingress:
+  - action: Allow
+  egress:
+  - action: Allow
+EOF
+```
+
+### 4. Allow traffic to Nginx from Busybox
+
+Create another Calico Network Policy which allows traffic from the nginx pod from the busybox "access" pod.
+
+```bash
+calicoctl create -f - <<EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: allow-busybox-nginx
+  namespace: advanced-policy-demo
+spec:
+  selector: all()
+  types:
+  - Ingress
+  - Egress
+  ingress:
+  - action: Allow
+    source:
+      selector: run == 'access'
+    destination:
+      selector: app == 'nginx'
+  egress:
+  - action: Allow
+EOF
+```
+
+#### Verify access - allowed traffic to nginx from "access" pod
+
+Now run the command to verify that we can access the nginx service.
+
+```bash
+wget -q --timeout=5 nginx -O -
+```
+
+It will return the HTML of the nginx welcome page.
+
+```
+<!DOCTYPE html>
+<html>
+<head>
+<title>Welcome to nginx!</title>...
+```
+
+{: .no-select-button}
+
+Next, try to retrieve the home page of google.com.
+
+```bash
+wget -q --timeout=5 google.com -O -
+```
+
+It returns:
+
+```
+wget: download timed out
+```
+
+{: .no-select-button}
+
+Access to `google.com` times out because we have not allowed that in our network policy the default deny still applies here, only the traffic that we have specified through policies are allowed through.
+
+## 7. Clean up namespace
```

Comment:
The clean up section needs to delete all the resources created during the tutorial.  That includes the global network policies created earlier.

Reply:
Done thank you!! :D